test/compilable/fix20964.d

```diff
@@ -0,0 +1,12 @@
+// https://issues.dlang.org/show_bug.cgi?id=20964
+
+uint foo(uint[3] m)
+{
+    auto p = m.ptr;
+    p += 3; // ok, p refers to the next right element after the block!
+    p -= 4; // ok, p refers to the next left element before the block!
```

Comment:
You need to change this to

```d
p -= 1;
```

in order for this file to compile.

Reply:
No he doesn't.
he wants to point 1 before the pointer.
((m.ptr +3) - 4) == m.ptr - 1

Reply:
Ahh, sorry.